easybuild/easyconfigs/s/SoX/SoX-14.4.2-GCC-11.3.0.eb

```diff
@@ -0,0 +1,34 @@
+# Institution: IT4Innovations National Supercomputing Center, Czech Republic
+# Author: Jakub Kropacek
+# License: GPLv3
+# Year: 2022
+
+easyblock = 'ConfigureMake'
+
+name = 'SoX'
+version = '14.4.2'
+
+homepage = 'http://sox.sourceforge.net/'
+docurls = 'http://sox.sourceforge.net/Docs/Documentation'
+description = """Sound eXchange, the Swiss Army knife of audio manipulation"""
+
+toolchain = {'name': 'GCC', 'version': '11.3.0'}
```

Comment:
I would suspect that any potential performance difference is just in the dependencies anyway, and they are all at GCCcore, so we can probably move this as well.

Reply:
Installation on one of our clusters failed during the compilation. By examining the logs I discovered that `SoX` is searching for usable `ld.gold`, and that the configuration fails because some of our systems (and I expect that others' too) still run RHEL7; using system `ld.gold` will therefore fail:
```
configure:3683: checking whether the C compiler works
configure:3705: gcc -O2 -ftree-vectorize -march=native -fno-math-errno -I/apps/all/FFmpeg/5.0.1-GCCcore-11.3.0/include -I/apps/all/libvorbis/1.3.7-GCCcore-11.3.0/include -I/apps/all/LAME/3.100-GCCcore-11.3.0/include -I/apps/all/FLAC/1.3.4-GCCcore-11.3.0/include -L/apps/all/FFmpeg/5.0.1-GCCcore-11.3.0/lib64 -L/apps/all/FFmpeg/5.0.1-GCCcore-11.3.0/lib -L/apps/all/libvorbis/1.3.7-GCCcore-11.3.0/lib64 -L/apps/all/libvorbis/1.3.7-GCCcore-11.3.0/lib -L/apps/all/LAME/3.100-GCCcore-11.3.0/lib64 -L/apps/all/LAME/3.100-GCCcore-11.3.0/lib -L/apps/all/FLAC/1.3.4-GCCcore-11.3.0/lib64 -L/apps/all/FLAC/1.3.4-GCCcore-11.3.0/lib -L/apps/all/GCCcore/11.3.0/lib64 -L/apps/all/GCCcore/11.3.0/lib conftest.c -lm -lpthread >&5
/usr/bin/ld.gold: --push-state: unknown option
/usr/bin/ld.gold: use the --help option for usage information
collect2: error: ld returned 1 exit status
configure:3709: $? = 1
configure:3747: result: no
```
However, this is fixed in newer releases of `binutils`. I would therefore vote for using the full `GCC` toolchain, instead of just `GCCcore`, for better portability.

Reply:
@anselmicz You should be able to work around that by still including `binutils` in the `builddependencies`, see for instance: https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/f/FLAC/FLAC-1.3.4-GCCcore-11.3.0.eb. The advantage of this approach is that tools built with `GCCcore` can also be used by the `intel` toolchain.

Reply:
That is true, I forgot about that! And it seems to be working (at least on my end). I added a new commit with suggested changes.